src/core_plugins/i18n/server/i18n/i18n.js

```diff
@@ -0,0 +1,122 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let i18nConfig = null;
+let registeredTranslations = {};
+
+/**
+ * Return all translations registered for the default locale.
+ * @return {Promise} - A Promise object which will contain on resolve a JSON object of all registered translations
+ */
+export function getTranslationsForDefaultLocale() {
+  let defaultLocale = '';
+  try {
+    defaultLocale = i18nConfig.get('i18n.locale');
+  } catch (e) {
+    defaultLocale = 'en';
+  }
+  return getTranslationsForLocale(defaultLocale);
+};
+
+/**
+ * Returns list of all registered locales.
+ * @return {Array} - Registered locales
+ */
+export function getRegisteredTranslationLocales() {
+  return Object.keys(registeredTranslations);
+};
+
+/**
+ * Return translations for a suitable locale from a user side locale list
+ * @param {Array<Sring>} languageTags -  BCP 47 language tags. The tags are listed in priority order as set in the Accept-Language header.
+ * @return {Object} - A Promise object which will contain on resolve a JSON object.
+ * This object will contain all registered translations for the highest priority locale which is registered with the i18n module.
+ * This object can be empty if no locale in the language tags can be matched against the registered locales.
+ */
+export function getTranslations(languageTags) {
+  let locale = '';
+
+  if (!_.isEmpty(languageTags)) {
+    const registeredLocales = getRegisteredTranslationLocales();
+    locale = getTranslationLocaleExactMatch(languageTags, registeredLocales) ||
+      getTranslationLocaleBestCaseMatch(languageTags, registeredLocales) || '';
```

Comment:
Following clarification with @Bargs and @srl295 last night, we have decided to follow the priority list that a user browser is configured for where there maybe a locale translation available but might not be the exact match with regard to the locale code and/or script. An example of this is that the highest priority locale of the user is 'en-US' but the locale translation available is 'en'. It is better select the 'en' locale rather than select the next highest locale which is an exact match.

The algorithm is now modified to check each user locale starting with the highest priority first for an exact match and then for best case match. If no match it then moves to the next user locale with the next highest priority.